packages/databyss-ui/components/PageContent/_helpers.ts

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect } from 'react'
+
+import { Block } from '@databyss-org/services/interfaces/'
+import { isAtomicInlineType } from '@databyss-org/editor/lib/util'
+import { EditorState } from '../../../databyss-editor/interfaces/index'
+import { stateBlockToHtmlHeader } from '@databyss-org/editor/lib/slateUtils.js'
+
+const getBlockPrefix = (type: string): string =>
+  ({
+    SOURCE: '@',
+    TOPIC: '#',
+  }[type])
+
+export const getPagePath = (page: EditorState): string => {
```

Comment:
I think this should return an array of strings representing the paths. That way, the rendering component can determine the padding between path elements and do truncation or ellipsis as needed.